src/java.desktop/unix/native/common/awt/X11Color.c

```diff
@@ -1224,11 +1224,6 @@ jobject awtJNI_GetColorModel(JNIEnv *env, AwtGraphicsConfigDataPtr aData)
             (*env)->PopLocalFrame(env, 0);
             return NULL;
         }
-
-        /* Set pData field of ColorModel to point to ColorData */
-        JNU_SetLongFieldFromPtr(env, awt_colormodel, g_CMpDataID,
-                                aData->color_data);
-
```

Comment:
This is the one that is more work to track down but it seems it belongs to the X11GraphicsConfig and is managed / disposed by that.